File: Fundamentals_of_Software_Design/1_Basic_skills/Угадывание числа/main.py
def guess_number():
    print("Загадайте число от 1 до 100.")
    low = 1
    high = 100

    while True:
        mid = (low + high) // 2
        answer = input(f"Загаданное число больше или равно {mid}? (д/н): ").strip().lower()

        if answer == "д":
            low = mid
        elif answer == "н":
            high = mid - 1
        else:
            print("Некорректный ввод, пожалуйста, введите 'д' или 'н'.")
            continue

        if low == high:
            print(f"Загаданное число - {low}")
            break

def main():
    while True:
        guess_number()
        play_again = input("Ещё один раунд? (д/н): ").strip().lower()
        if play_again != "д":
            print("Спасибо за игру!")
            break

if __name__ == "__main__":
    main()

File: Fundamentals_of_Software_Design/1_Basic_skills/Угадывание числа/main.c
#include <stdio.h>
#include <stdlib.h>

void guess_number() {
    int low = 1, high = 100, mid;
    char answer;

    printf("Загадайте число от 1 до 100.\n");

    while (low <= high) {
        mid = (low + high) / 2;
        printf("Загаданное число больше или равно %d? (д/н): ", mid);
        scanf(" %c", &answer);

        if (answer == 'д') {
            low = mid;
        } else if (answer == 'н') {
            high = mid - 1;
        } else {
            printf("Некорректный ввод, пожалуйста, введите 'д' или 'н'.\n");
            continue;
        }

        if (low == high) {
            printf("Загаданное число - %d\n", low);
            break;
        }
    }
}

int main() {
    char play_again;

    do {
        guess_number();
        printf("Ещё один раунд? (д/н): ");
        scanf(" %c", &play_again);
    } while (play_again == 'д');

    printf("Спасибо за игру!\n");
    return 0;
}
